### DISCUSSION TITLE
command line "magick" vs "magick convert" in documentation

### DISCUSSION THREAD

**Discussion by safetycar** [ImageMagick/ImageMagick#5279] [General] [answered]
I keep reading around here that there are differences between `magick` and `magick convert`.

But I've been trying to find this differences in the documentation more than once and never reached anything that made me understand it properly.

For example I go to de documentation and the closest thing I find are this 2 links:
https://imagemagick.org/script/command-line-tools.php
![2022-06-26 10_46_09-ImageMagick – Command-line Tools — Mozilla Firefox](https://user-images.githubusercontent.com/536820/175807055-38dbff76-5cbc-4ff6-99aa-d4fb19da7013.png)

The names are the same in the page but the links are different.
One points to "/script/magick.php" and other to "/script/convert.php"

In the effort to find the differences I've found small things.
There are others but what I've noticed the most are this two.

1. I'm not sure 100% what it tries to say but what I'd understand is that both things would be the same so quite the opposite of what I've read here.
![2022-06-26 10_53_56-ImageMagick – Command-line Tools_ Magick — Mozilla Firefox](https://user-images.githubusercontent.com/536820/175807145-3b4cd013-f58e-4aca-9ae2-480523fe8c36.png)

2. And this:
![2022-06-26 10_54_43-ImageMagick – Command-line Tools_ Magick — Mozilla Firefox](https://user-images.githubusercontent.com/536820/175807093-ba27abc5-fd77-4023-b2a5-d551fa94d96a.png)
Which doesn't actually seem to exist.

Is there somewhere else that I should be looking for it?

**Comment by snibgo** [comment, marked as answer]
"magick" by itself uses IM v7 syntax. "magick convert" uses the older v6 syntax.

Differences between v6 and v7 syntax are shown on https://www.imagemagick.org/script/porting.php .

V7 syntax is better than v6. It is more powerful, with new features, and has improved error-checking. I suggest that "magick convert" is used *only* when if you have an old v6 script that doesn't work under v7.

EDIT: Put it another way: "magick convert" is provided for backwards compatibility. For new work, "magick" should be used.

**Comment by safetycar** [comment]
I'm wondering where did you find that link?

Also there I can read this:
> these utilities are symbolic links to the magick utility. You can also invoke them from the magick utility

It's not that I don't believe that they are different, but the docs are a bit weird about it.

**Comment by snibgo** [comment]
I found that link on my personal local homepage. I expect there is a link to it somewhere on the official documentation.

Your installation may have a symbolic link so that the "convert" command executes the same file as the "magick" command. Your operating system can tell you if this is the case. The "magick" program knows (through "argv[0]") whether is was called from a "convert" or "magick" command, and will use the v6 or v7 syntax.

**Comment by safetycar** [comment]
Ok, thanks. I think I'm more convinced now after that explanation and going back to that "porting" link where there are some of the differences.
Although maybe some of the things I mentioned along the way would be good to be reviewed for other people.
